### PR TITLE
Sponge file init patch

### PR DESCRIPTION
**Pull request by MJHarrison-GFDL**
This addresses #1218.

### CODE REVIEW

File: src/initialization/MOM_state_initialization.F90 [hunk 1]
```diff
@@ -1821,73 +1826,82 @@ subroutine initialize_sponges_file(G, GV, US, use_temperature, tv, param_file, C
...
+      deallocate(h)
+    else
+      ! Initialize sponges without supplying sponge grid
+      call initialize_ALE_sponge(Idamp, G, param_file, ALE_CSp)
+    endif
+
+
+    ! The remaining calls to set_up_sponge_field can be in any order.
+    if ( use_temperature) then
+      call set_up_ALE_sponge_field(filename, potemp_var, Time, G, GV, US, tv%T, ALE_CSp)
```
Comment (Hallberg-NOAA) [inline src/initialization/MOM_state_initialization.F90:1898]: set_up_ALE_sponge_field is an overloaded interface to two different routines. I would have thought that the two different interfaces should have been used for the two settings of time_space_interp_sponge, but both options are using the variant that should be used with time_space_interp_sponge=.true.

  Given that this code was not working before, in a way that our testing was not detecting, I think that we need to be particularly careful in implementing these corrections. As a confidence-building measure, I would prefer that this be implemented in the simpler way, using the static target value interface when time_space_interp_sponge=.false. Once this has been demonstrated to work (e.g., the MOM6-issue that triggered all this has been closed), we could then try using the same interface for both variants and verifying that they give identical answers in test cases that exercise that option.

**Comment by MJHarrison-GFDL** [comment]
@Hallberg-NOAA That sounds like a good plan. I will modify the tc4 test accordingly.

File: src/initialization/MOM_state_initialization.F90 (hunk 1 above)
Comment (MJHarrison-GFDL) [reply src/initialization/MOM_state_initialization.F90]: @Hallberg-NOAA . Updated as suggested.

Comment (Hallberg-NOAA) [reply src/initialization/MOM_state_initialization.F90]: I am now satisfied that with these latest modifications this PR is very likely to be exactly what we need to address and close Issue #1218.

**Review by Hallberg-NOAA — APPROVED** [review]
I am now satisfied that these are the right changes to make to address Issue #1218, and I am approving this PR.

**Comment by marshallward** [comment]
https://gitlab.gfdl.noaa.gov/ogrp/MOM6/-/pipelines/11391

**Pull request merged**